### PR TITLE
HP-2372: Replace APINotImplementedError with ProfileAlreadyExistsForUserError

### PR DESCRIPTION
**Pull request by danipran**
The TODO for OM-385/HP-2368 is obsolete, so replace the `APINotImplementedError` with something more descriptive (since it won't be implemented), i.e. `ProfileAlreadyExistsForUserError`. `APINotImplementedError` is not used anymore after this, so remove it as well.

Also remove another obsolete TODO (HP-2370).

### CODE REVIEW

File: profiles/schema.py [hunk 1]
```diff
@@ -1862,7 +1860,7 @@ class Mutation(graphene.ObjectType):
         "an error.\n\n"
         "Possible error codes:\n\n"
         "* `PROFILE_MUST_HAVE_PRIMARY_EMAIL`: If trying to get rid of the profile's primary email.\n"
-        "* `API_NOT_IMPLEMENTED_ERROR`: Returned if the currently authenticated user already has a profile."
+        "* `PROFILE_DOES_NOT_EXIST_ERROR`: Returned if the currently authenticated user already has a profile."
```
Comment (charn) [inline profiles/schema.py:1863]:
```suggestion
        "* `PROFILE_ALREADY_EXISTS_FOR_USER_ERROR`: Returned if the currently authenticated user already has a profile."
```

File: open_city_profile/consts.py [hunk 2]
```diff
@@ -9,6 +8,7 @@
 VALIDATION_ERROR = "VALIDATION_ERROR"
 JWT_AUTHENTICATION_ERROR = "JWT_AUTHENTICATION_ERROR"
 DATA_CONFLICT_ERROR = "DATA_CONFLICT_ERROR"
+PROFILE_ALREADY_EXISTS_FOR_USER_ERROR = "PROFILE_ALREADY_EXISTS_FOR_USER_ERROR"
```
Comment (charn) [inline open_city_profile/consts.py:11]: Maybe move this under `# Profile specific errors`

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-API branch is deployed to platta: https://helsinki-profile-pr485.api.dev.hel.ninja** :rocket::rocket::rocket:

File: open_city_profile/views.py [hunk 3]
```diff
@@ -52,12 +52,12 @@
     ObjectDoesNotExist: OBJECT_DOES_NOT_EXIST_ERROR,
     TokenExpiredError: TOKEN_EXPIRED_ERROR,
     PermissionDenied: PERMISSION_DENIED_ERROR,
-    APINotImplementedError: API_NOT_IMPLEMENTED_ERROR,
     ValidationError: VALIDATION_ERROR,
     graphene_validator.errors.ValidationGraphQLError: VALIDATION_ERROR,
     InvalidEmailFormatError: INVALID_EMAIL_FORMAT_ERROR,
     AuthenticationError: JWT_AUTHENTICATION_ERROR,
     DataConflictError: DATA_CONFLICT_ERROR,
+    ProfileAlreadyExistsForUserError: PROFILE_ALREADY_EXISTS_FOR_USER_ERROR,
```
Comment (charn) [inline open_city_profile/views.py:60]: And also this under `error_codes_profile`

**Review by charn — APPROVED** [review]
LGTM! :+1: :shipit:

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-API branch is deployed to platta: https://helsinki-profile-pr485.api.dev.hel.ninja** :rocket::rocket::rocket:

**Pull request merged**